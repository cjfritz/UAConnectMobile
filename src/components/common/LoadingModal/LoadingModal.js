import React from 'react';
import { View } from 'react-native';
import Modal from 'react-native-root-modal';
import { Spinner, H3 } from 'native-base';
import styles from './LoadingModal.style';

export const LoadingModal = ({ visible, loadingLabel }) => visible && (
  <Modal style={ styles.modalStyle } visible={ visible }>
    <View style={ styles.container }>
      <Spinner style={ styles.spinner } />
      <H3 style={ styles.text }>{ loadingLabel }</H3>
    </View>
  </Modal>
);
