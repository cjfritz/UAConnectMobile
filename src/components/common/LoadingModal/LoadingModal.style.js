import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    alignItems: 'center',
    backgroundColor: 'rgba(0, 0, 0, 0.5)',
    bottom: 0,
    justifyContent: 'center',
    left: 0,
    position: 'absolute',
    right: 0,
    top: 0,
  },
  modalStyle: {
    bottom: 0,
    left: 0,
    position: 'absolute',
    right: 0,
    top: 0,
  },
  text: {
    color: 'white',
  },
});
