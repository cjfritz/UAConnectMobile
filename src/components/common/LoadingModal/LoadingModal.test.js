import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { LoadingModal } from './LoadingModal';

describe('<LoadingModal />', () => {
  it('cannot render with default props', () => {
    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<LoadingModal />);
    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
  });

  it('cannot render without visibility', () => {
    const props = {
      visible: false,
      loadingLabel: null,
    };

    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<LoadingModal { ...props } />);
    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
  });

  it('can render with visibility', () => {
    const props = {
      visible: true,
      loadingLabel: 'This is a test',
    };

    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<LoadingModal { ...props } />);
    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
  });

  it('can render with visibility and without loadingLabel', () => {
    const props = {
      visible: true,
      loadingLabel: undefined,
    };

    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<LoadingModal { ...props } />);
    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
  });
});
