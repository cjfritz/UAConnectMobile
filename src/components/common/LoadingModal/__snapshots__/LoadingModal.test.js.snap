// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingModal /> can render with visibility 1`] = `
<Modal
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
  visible={true}
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.5)",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Component
      animating={true}
      color="white"
      hidesWhenStopped={true}
      size="large"
    />
    <Styled(H3)
      style={
        Object {
          "color": "white",
        }
      }
    >
      This is a test
    </Styled(H3)>
  </Component>
</Modal>
`;

exports[`<LoadingModal /> can render with visibility and without loadingLabel 1`] = `
<Modal
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
  visible={true}
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.5)",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Component
      animating={true}
      color="white"
      hidesWhenStopped={true}
      size="large"
    />
    <Styled(H3)
      style={
        Object {
          "color": "white",
        }
      }
    />
  </Component>
</Modal>
`;

exports[`<LoadingModal /> cannot render with default props 1`] = `undefined`;

exports[`<LoadingModal /> cannot render without visibility 1`] = `false`;
